### ISSUE TITLE
open_rasterio: Need to move Affine object to left side

### ISSUE THREAD

**Issue by snowman2** [pydata/xarray#3449]
#### MCVE Code Sample
```python
xarray.open_rasterio(...)
```

#### Problem Description

Need to move the `transform` to the left side.

Warning:
```
DeprecationWarning: Right multiplication will be prohibited in version 3.0
    x, _ = (np.arange(nx) + 0.5, np.zeros(nx) + 0.5) * transform
```

Should be:
```
 x, _ = transform * (np.arange(nx) + 0.5, np.zeros(nx) + 0.5)
```

**Comment by dcherian** [comment]
Thanks. This is fixed on master by #3428

**Comment by snowman2** [comment]
Nice!
